programs/bpf_loader/src/lib.rs

```diff
@@ -298,6 +298,20 @@ fn check_loader_id(id: &Pubkey) -> bool {
         || bpf_loader_upgradeable::check_id(id)
 }
 
+fn calculate_heap_cost(heap_size: u64, heap_cost: u64, enable_rounding_fix: bool) -> u64 {
+    const KILOBYTE: u64 = 1024;
+    const PAGE_SIZE_KB: u64 = 32;
+    let mut rounded_heap_size = heap_size;
+    if enable_rounding_fix {
+        rounded_heap_size = rounded_heap_size
+            .saturating_add(PAGE_SIZE_KB.saturating_sub(1).saturating_mul(KILOBYTE));
+    }
+    rounded_heap_size
+        .saturating_div(PAGE_SIZE_KB.saturating_mul(KILOBYTE))
+        .saturating_sub(1)
```

Comment:
also, why always subtract one? That way the first 31 KiB don't count.

Reply:
gotit

Reply:
To keep compatible with the original behavior it should be the following, right?
`.saturating_sub(if enable_rounding_fix { 0 } else { 1 })`

Reply:
Mmm, don't think need to do that. The first 31KiB costs zero in both before/after activation. The test at Ln3945 asserts these scenarios.

Reply:
If it is intentional then leave it as is.